planning/behavior_path_planner/src/scene_module/avoidance/avoidance_utils.cpp

```diff
@@ -342,4 +342,149 @@ Polygon2d createEnvelopePolygon(
+    }
+  }
+
+  return {first_point, second_point};
+}
+
+std::vector<PolygonPoint> getEdgePoints(
+  const std::vector<Point> & bound, const std::vector<Point> & edge_points,
+  const double lat_dist_to_path, size_t & start_segment_idx, size_t & end_segment_idx)
+{
+  std::vector<PolygonPoint> edge_points_data;
+  for (const auto & edge_point : edge_points) {
+    const size_t segment_idx = motion_utils::findNearestSegmentIndex(bound, edge_point);
+    start_segment_idx = std::min(start_segment_idx, segment_idx);
+    end_segment_idx = std::max(end_segment_idx, segment_idx);
+
+    PolygonPoint edge_point_data;
+    edge_point_data.point = edge_point;
+    edge_point_data.lat_dist_to_bound = motion_utils::calcLateralOffset(bound, edge_point);
+    edge_point_data.lon_dist = motion_utils::calcSignedArcLength(bound, 0, edge_point);
+    if (lat_dist_to_path >= 0.0 && edge_point_data.lat_dist_to_bound > 0.0) {
+      continue;
+    } else if (lat_dist_to_path < 0.0 && edge_point_data.lat_dist_to_bound < 0.0) {
+      continue;
+    }
+
+    edge_points_data.push_back(edge_point_data);
+  }
+
+  return edge_points_data;
+}
+
+std::vector<Point> updateBoundary(
+  const std::vector<Point> & original_bound, const std::vector<PolygonPoint> & points,
+  const size_t start_segment_idx, const size_t end_segment_idx)
+{
+  if (start_segment_idx >= end_segment_idx) {
+    return original_bound;
+  }
+
+  std::vector<Point> updated_bound;
+  for (size_t i = 0; i <= start_segment_idx; ++i) {
+    updated_bound.push_back(original_bound.at(i));
+  }
+  for (size_t i = 0; i < points.size(); ++i) {
+    updated_bound.push_back(points.at(i).point);
+  }
+  for (size_t i = end_segment_idx + 1; i < original_bound.size(); ++i) {
+    updated_bound.push_back(original_bound.at(i));
+  }
+
+  return updated_bound;
+}
+
+void generateDrivableArea(
+  PathWithLaneId & path, const std::vector<DrivableLanes> & lanes, const double vehicle_length,
+  const std::shared_ptr<const PlannerData> planner_data, const ObjectDataArray & objects,
+  const bool enable_bound_clipping)
+{
+  util::generateDrivableArea(path, lanes, vehicle_length, planner_data);
+  if (objects.empty() || !enable_bound_clipping) {
+    return;
+  }
+
+  path.left_bound = motion_utils::resamplePointVector(path.left_bound, 1.0, true);
+  path.right_bound = motion_utils::resamplePointVector(path.right_bound, 1.0, true);
+
+  for (const auto & object : objects) {
+    const auto & obj_pose = object.object.kinematics.initial_pose_with_covariance.pose;
+    const auto & obj_poly = object.envelope_poly;
+    constexpr double threshold = 0.01;
+    const auto edge_points = getEdgePoints(obj_poly, threshold);
+
+    // get boundary
+    const double lat_dist_to_path = motion_utils::calcLateralOffset(path.points, obj_pose.position);
+    auto & bound = lat_dist_to_path < 0.0 ? path.right_bound : path.left_bound;
+
+    // update boundary
+    size_t start_segment_idx = bound.size() == 1 ? 0 : bound.size() - 2;
```

Comment:
It is better to use () because developer may misread
```suggestion
    size_t start_segment_idx = (bound.size() == 1 ? 0 : (bound.size() - 2));
```

Reply:
done